### PR TITLE
GitHub action to run periodically or by trigger to extractor

### LINKED ISSUE #143: GitHub action to run periodically or by trigger to extract configs
For now we just need to run `./gradlew build` and commit and push to main branch all channels.

But this can be done as GitHub action, to be done daily.

Example on how to do this:
https://github.com/checkstyle/checkstyle/blob/977931a4c52791d55adce728106eef3766502e12/.github/workflows/release-maven-prepare.yml#L7-L20

https://github.com/checkstyle/checkstyle/blob/977931a4c52791d55adce728106eef3766502e12/.github/workflows/release-maven-prepare.yml#L43-L47

https://github.com/checkstyle/checkstyle/blob/977931a4c52791d55adce728106eef3766502e12/.github/workflows/release-maven-prepare.yml#L52-L54

### PR DESCRIPTION
**Pull request by relentless-pursuit**
Resolves #143

### CODE REVIEW

**Comment by relentless-pursuit** [comment]
And, how to test this CI?

**Review by romani — CHANGES_REQUESTED** [review]
Create new workflow that will do only extract and commit

**Comment by relentless-pursuit** [comment]
Testing of the workflow: https://github.com/checkstyle-gsoc/test-configs/actions/runs/10396815739

@Romani

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,33 +1,50 @@
-name: Java CI
+name: Java CI and Automated Extraction
```
Comment (romani) [inline .github/workflows/ci.yml:1]: Separate workflow please

**Review by romani — CHANGES_REQUESTED** [review]
Items

**Pull request closed without merging**